### PR TITLE
Drop currentModId as a concept

### PR DESCRIPTION
**Pull request by grahamlangford**
## What does this PR do?

- Refactors to eliminate the need for `currentModId`
- No longer set `activeModId` to null when selecting a modComponentId

## Discussion

- See the InvariantMiddleware

For more information on our expectations for the PR process, see the
[code review principles doc](https://www.notion.so/pixiebrix/Code-Review-Principles-1ce7276b82a84d2a995d55ad85e1310d?pvs=4)

### CODE REVIEW

File: applications/browser-extension/src/pageEditor/store/editor/editorInvariantMiddleware.ts [hunk 1]
```diff
@@ -46,19 +47,34 @@ class InvariantViolationError extends Error {
...
+  dispatch: AppDispatch,
+): void {
   // Assert that a mod and mod component item cannot be selected at the same time
-  if (selectActiveModId(state) && selectActiveModComponentId(state)) {
+  const activeModId = selectActiveModId(state);
+  const activeModComponentId = selectActiveModComponentFormState(state);
+
+  if (activeModComponentId && !activeModId) {
+    dispatch(actions.setActiveModId(activeModComponentId.modMetadata.id));
+  }
```
Comment (grahamlangford) [inline applications/browser-extension/src/pageEditor/store/editor/editorInvariantMiddleware.ts:60]: This should only ever occur the first time the Page Editor is accessed after this is merged. Alternatively, I could leave it and use a "migration" to reset the synced state

Comment (twschiller) [reply applications/browser-extension/src/pageEditor/store/editor/editorInvariantMiddleware.ts:60]: > This should only ever occur the first time the Page Editor is accessed after this is merged. Alternatively, I could leave it and use a "migration" to reset the synced state

IMO, I'd prefer the migration. Or at the very least we need a disclaimer explaining why this line of code is here / and the context for when the condition would fire

**Review by twschiller — APPROVED** [review]
Looking good overall. I'd prefer a migration: https://github.com/pixiebrix/pixiebrix-extension/pull/9490/files#r1837109301

See failing `types` job for unused import

File: applications/browser-extension/src/pageEditor/store/editor/editorSelectors/editorPipelineSelectors.ts
Comment (grahamlangford) [inline applications/browser-extension/src/pageEditor/store/editor/editorSelectors/editorPipelineSelectors.ts:1]: Unrelated, just noticed while debugging

**Pull request merged**